### PR TITLE
(olap) (bugfix) fix hll-oom issue

### PR DESCRIPTION
**Pull request by zuochunwei**
# Proposed changes
there is a oom problem reported by caiconghui001, but i can not find the bug by checking code carefully.
it's difficult to make it re-appear, so i decide to rewrite it, the thought is refer to other implements.

by comparing test, the new implement is more faster and memory save.  

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zenoyang** [comment]
I used a table with 20 million rows to test the query. The current master branch version query takes 18s. After optimization, it only takes 4s. The sql is as follows: (1FE, 3BE)

```sql
select 
    dt, HLL_UNION_AGG(union_id)
from
    xxx
where
    dt = 20211130 and hour <= 19 and minute > 10
group by dt;
```

I started 10 concurrent queries for stress testing, and found that the memory consumption was also reduced after optimization, as shown in the following figure:
image

File: be/src/olap/hll.cpp [hunk 1]
```diff
@@ -418,33 +353,32 @@ int64_t HyperLogLog::estimate_cardinality() const {
...
-        if (_registers[i] == 0) {
+        if (_registers.data[i] == 0) {
             ++num_zero_registers;
         }
     }
 
     harmonic_mean = 1.0f / harmonic_mean;
     double estimate = alpha * num_streams * num_streams * harmonic_mean;
-    // according to HyperLogLog current correction, if E is cardinal
+    // according to HerperLogLog current correction, if E is cardinal
```
Comment (morningman) [inline be/src/olap/hll.cpp:365]:
```suggestion
    // according to HyperLogLog current correction, if E is cardinal
```

**Comment by zbtzbtzbt** [comment]
cool ! 
I am writing `SSE2` version for `_merge_registers` which will help machines that not support `AVX2`

**Comment by zbtzbtzbt** [comment]
```
#elif __SSE2__
        int loop = HLL_REGISTERS_COUNT / 16; // 16 = 128/8
        uint8_t* dst = _registers.data;
        const uint8_t* src = other_registers;
        for (int i = 0; i < loop; i++) {
            __m128i xa = _mm_load_si128((const __m128i*)dst);
            __m128i xb = _mm_load_si128((const __m128i*)src);
            _mm_storeu_si128((__m128i*)dst, _mm_max_epu8(xa, xb));
            src += 16;
            dst += 16;
        }
#else
```

**Comment by zbtzbtzbt** [comment]
When inserting HLL column data  in following way, it will coredump. @zuochunwei 

`insert into data_zuo_oom_hll_master_41a15ccd select from data_tmp a where a.dt = 20210330;`
(SQL is trimmed)

```
*** Aborted at 1648013851 (unix time) try "date -d @1648013851" if you are using GNU date ***
PC: @     0x55a98e4c4a9f doris::MemTable::RowCursorComparator::operator()()
*** SIGSEGV (@0x0) received by PID 331999 (TID 0x7faaceee1700) from PID 0; stack trace: ***
    @     0x55a990215592 google::(anonymous namespace)::FailureSignalHandler()
    @     0x7fab5f61c400 (unknown)
    @     0x55a98e4c4a9f doris::MemTable::RowCursorComparator::operator()()
    @     0x55a98e4c5a17 doris::MemTable::insert()
    @     0x55a98ede69aa doris::DeltaWriter::write()
    @     0x55a98e5fdfcc doris::TabletsChannel::add_batch()
    @     0x55a98e5fa4ef doris::LoadChannel::add_batch()
    @     0x55a98e5f4393 doris::LoadChannelMgr::add_batch()
    @     0x55a98e6a528b _ZZN5doris20PInternalServiceImplINS_15PBackendServiceEE23tablet_writer_add_batchEPN6google8protobuf13RpcControllerEPKNS_28PTabletWriterAddBatchRequestEPNS_27PTabletWriterAddBatchResultEPNS4_7ClosureEENKUlvE_clEv
    @     0x55a98e5939f3 doris::PriorityThreadPool::work_thread()
    @     0x55a9922bd350 execute_native_thread_routine
    @     0x7fab5f3d1ea5 start_thread
    @     0x7fab5f6e49fd __clone
```

**Pull request closed without merging**